pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,131 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	ApiVip            string `json:"apiVip,omitempty"`
+	ApiVipDnsName     string `json:"apiVipDnsName,omitempty"`
+	IngressVip        string `json:"ingresVip,omitempty"`
+	VipDhcpAllocation bool   `json:"vipDhcpAllocation,omitempty"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	*Networking `json:"networking,omitempty"`
+
+	// Proxy defines the proxy settings for the cluster.
+	// If unset, the cluster will not be configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
```

Comment:
Similar for proxy, now identical to install config.

Reply:
After today's call, I'm going to drop this, it sounds like this may move to a new InstallEnvironment, tied to the data needed to get an ISO created for a cluster. 

@avishayt @filanov when you add this, could you re-use this structure which matches the installconfig in openshift-installer:

```go
// Proxy defines the proxy settings for the cluster.    
// At least one of HTTPProxy or HTTPSProxy is required.    
type Proxy struct {    
    // HTTPProxy is the URL of the proxy for HTTP requests.    
    // +optional    
    HTTPProxy string `json:"httpProxy,omitempty"`    
    
    // HTTPSProxy is the URL of the proxy for HTTPS requests.    
    // +optional    
    HTTPSProxy string `json:"httpsProxy,omitempty"`    
    
    // NoProxy is a comma-separated list of domains and CIDRs for which the proxy should not be used.    
    // +optional    
    NoProxy string `json:"noProxy,omitempty"`    
}    
```

Reply:
Removed as we expect to go to InstallEnvironment CRD.